### DISCUSSION TITLE
What am i doing wrong? - Git branches

### DISCUSSION THREAD

**Discussion by deniscooper** [community/community#31209] [New to GitHub]
Hi,

not sure what im doing wrong here. I am using vscode and got a repo where I have created a branch and been making changes in the branch. I switched back to main, created a new branch and made another change, i committed that change then merged it to the main branch, but not my main branch contains code from the first branch i was working on.

**Comment by ihnatenkoo** [comment]
Hello, you need switch to the first branch and merge it to the main branch

**Comment by airtower-luna** [comment]
That sounds like you might have actually started the second branch from the first one instead of `main`, but it's impossible to be sure without knowing exactly what you did and seeing the commits & branches.

**Comment by deniscooper** [comment]
i think i figured it out, i hadn't committed my work in the first branch before switching and creating a new branch from main. So I managed to revert back the commits and commit and switch in the right order
